light/provider/http/http.go

```diff
@@ -100,10 +100,16 @@ func NewWithClientAndOptions(chainID string, client rpcclient.RemoteClient, opti
 	}
 }
 
-func (p *http) String() string {
+func (p *http) ID() string {
 	return fmt.Sprintf("http{%s}", p.client.Remote())
 }
 
+// Ideally we want only one ID function. String is here to not break the Provider interface with another method
+// which would cause changes to BlockProvider in the internal/statesync
```

Comment:
It would only be helpful for someone to know what we expect from this function which is right now not a sufficient reason to introduce this change to other packages (my opinion).  We just need to state it clearly in the doc.